### ISSUE TITLE
Too many CI failures

### ISSUE THREAD

**Issue by paulp** [quasar-analytics/quasar#1295]
I wish to suggest that the spurious failure rate on travis is too high - especially when all four jobs must succeed. The damage it does to one's flow to have to perpetually check back in on factors out of one's control in order to make forward progress should not be underestimated.

[Here's my most recent example](https://travis-ci.org/quasar-analytics/quasar/builds/147578627). Three out of four jobs succeed, one fails because:

```
[error]     x be 400 if attempting to move a dir into a file
[error]      Gave up after only 1 passed tests. 10 tests were discarded. (DataServiceSpec.scala:54)
```

I'm not sure what to make of that actually - why did it give up after only 10 discards?

Here's a similar pattern from a different spurious failure:

```
[error]     x and list of children for existing nonempty directory
[error]      Gave up after only 1 passed tests. 10 tests were discarded. (MetadataServiceSpec.scala:65)
```

**Comment by mossprescott** [comment]
Yes, we're all very aware of how disruptive this is and we've made a couple of attempts to fix it with incomplete success.

It's somehow related to the behavior of scalacheck when `minTestsOK` is set to a small value, which we do in some cases because the tests otherwise run very slow.

**Comment by mossprescott** [comment]
For some context, see #1172. I believe there was another PR and a JIRA ticket, but I can't find either at the moment.

**Comment by paulp** [comment]
Right, I didn't mean to suggest you weren't aware. I'll see what I can figure out.

**Comment by paulp** [comment]
Collecting more examples for reference. Probably a completely separate underlying cause, failing right at the end:
```
 Quasar Version: 11.0.4
 Launching Quasar JAR...
 Quasar PID: 8076
 JarClassLoader: Warning: quasar/LowPriorityImplicits.class in lib/quasar-foundation-internal_2.11-11.0.4.jar is hidden by lib/quasar-core- internal_2.11-11.0.4.jar (with different bytecode)
 Read config from /home/travis/.config/quasar/quasar-config.json
 Killing Quasar JAR...
 Status code from killing Quasar: 0
 Quasar failed to start
```

**Comment by paulp** [comment]
```
[error]     x be 204 with existing dir
[error]      Gave up after only 1 passed tests. 10 tests were discarded. (DataServiceSpec.scala:54)
```

**Comment by paulp** [comment]
[An interesting one](https://travis-ci.org/quasar-analytics/quasar/jobs/147583994):
```
 *** glibc detected *** /usr/lib/jvm/java-8-oracle/bin/java: free(): invalid pointer: 0x0000000100ca7df8 ***
 ======= Backtrace: =========
 /lib/x86_64-linux-gnu/libc.so.6(+0x7da26)[0x7f1cb48d9a26]
 /usr/lib/jvm/java-8-oracle/jre/lib/amd64/server/libjvm.so(+0x8de889)[0x7f1cb41b4889]
 /usr/lib/jvm/java-8-oracle/jre/lib/amd64/server/libjvm.so(+0x606011)[0x7f1cb3edc011]
 /usr/lib/jvm/java-8-oracle/jre/lib/amd64/server/libjvm.so(+0x978050)[0x7f1cb424e050]
 /usr/lib/jvm/java-8-oracle/jre/lib/amd64/server/libjvm.so(+0x978d76)[0x7f1cb424ed76]
 /usr/lib/jvm/java-8-oracle/jre/lib/amd64/server/libjvm.so(+0xa812ef)[0x7f1cb43572ef]
 /usr/lib/jvm/java-8-oracle/jre/lib/amd64/server/libjvm.so(+0xa81770)[0x7f1cb4357770]
 /usr/lib/jvm/java-8-oracle/jre/lib/amd64/server/libjvm.so(+0x8e8158)[0x7f1cb41be158]
 /lib/x86_64-linux-gnu/libpthread.so.0(+0x7e9a)[0x7f1cb503ae9a]
 /lib/x86_64-linux-gnu/libc.so.6(clone+0x6d)[0x7f1cb494f36d]
 ======= Memory map: ========
 00400000-00401000 r-xp 00000000 fc:0d 50472032                           /usr/lib/jvm/java-8-oracle/bin/java
 00600000-00601000 rw-p 00000000 fc:0d 50472032                           /usr/lib/jvm/java-8-oracle/bin/java
 00a1a000-00a3b000 rw-p 00000000 00:00 0                                  [heap]
 80000000-103cc0000 rw-p 00000000 00:00 0
 103cc0000-140000000 ---p 00000000 00:00 0
...
```

**Comment by paulp** [comment]
A second hit on this one (and yes I realize there are diminishing returns in recording each failure.)
```
[error]     x and list of children for existing nonempty directory
[error]      Gave up after only 1 passed tests. 11 tests were discarded. (MetadataServiceSpec.scala:65)
```

**Comment by paulp** [comment]
With the disabling of the spark project we may have returned to the more everyday sort of spurious failures, which is great. [Here's one](https://travis-ci.org/paulp/quasar/jobs/149495901).
```
*[error]               x precise and in a single json array (application/json; mode="precise")
 [error]                Falsified after 38 passed tests.
 [error]                > ARG_0: SingleFileMemState(FileIn(DirIn(DirIn(DirIn(Root,DirName(ecksz)),DirName(5)),DirName(o)), FileName(3rS<U+0DE2>nhcj0ye%bejbIxth)),Vector(Dec(-897.0), Arr(List(Binary(Array[Byte](127, -59, 1, -48)), Null))))
 [error]                > '{ "error": { "status": "Path not found.", "detail": { "path": "/ecksz/5/o/3rS<U+0DE2>nhcj0ye%BE jbIxth" } } }'
 [error]
 [error]                 is not equal to
 [error]
 [error]                '[
 [error]                -897.0,
 [error]                [ { "$binary": "f8UB0A==" }, null ]
 [error]                ]
 [error]                ' (DataServiceSpec.scala:54)
 [error] Actual:   { "error": { "status": "Path not found.", "detail": { "path": "/ecksz/5/o/3rS<U+0DE2>nhcj0ye%BEjbIxth"  } } }
 [error] Expected: [
 [error] -897.0,
 [error] [ { "$binary": "f8UB0A==" }, null ]
 [error] ]
```

**Comment by wemrysi** [comment]
And another

```

[info] QueryFile should
[info]   descendantFiles
[error]     x returns all descendants of the given directory
[error]      Falsified after 42 passed tests.
[error]      > ARG_0: List("FileIn(DirIn(DirIn(Root,DirName(A)),DirName(.)),FileName(g))", "FileIn(DirIn(DirIn(Root,DirName(w)),DirName(f)),FileName(o))", "FileIn(DirIn(Root,DirName(f)),FileName(/gb))", "FileIn(DirIn(Root,DirName(wᒮ)),FileName(f))", "FileIn(DirIn(Root,DirName(f)),FileName(oe/))")
[error]      > ARG_1: NonEmpty[FileIn(Current,FileName(/)),FileIn(DirIn(Current,DirName(ii)),FileName(xa)),FileIn(DirIn(Current,DirName(.a)),FileName(.5)),FileIn(DirIn(Current,DirName(u)),FileName(o)),FileIn(Current,FileName(c)),FileIn(DirIn(Current,DirName(qb)),FileName(sl)),FileIn(Current,FileName(c)),FileIn(DirIn(DirIn(Current,DirName(h)),DirName(u)),FileName(蟋)),FileIn(Current,FileName(h)),FileIn(DirIn(DirIn(Current,DirName(o)),DirName(6)),FileName(z)),FileIn(Current,FileName(i))]
[error]      > ARG_2: DirIn(DirIn(Root,DirName(w)),DirName(nk))
[error]      > 'Right(Set(FileIn(DirIn(Current,DirName(u)),FileName(o)), FileIn(DirIn(Current,DirName(.a)),FileName(.5)), FileIn(DirIn(Current,DirName(ii)),FileName(xa)), FileIn(Current,FileName(/)), FileIn(DirIn(DirIn(Current,DirName(o)),DirName(6)),FileName(z)), FileIn(Current,FileName(h)), FileIn(Current,FileName(c)), FileIn(DirIn(DirIn(Current,DirName(h)),DirName(u)),FileName(蟋)), FileIn(DirIn(Current,DirName(qb)),FileName(sl)), FileIn(Current,FileName(i))))' is Right but Set(FileIn(DirIn(Current,DirName(u)),FileName(o)), FileIn(DirIn(Current,DirName(.a)),FileName(.5)), FileIn(DirIn(Current,DirName(ii)),FileName(xa)), FileIn(Current,FileName(/)), FileIn(DirIn(DirIn(Current,DirName(o)),DirName(6)),FileName(z)), FileIn(Current,FileName(h)), FileIn(Current,FileName(c)), FileIn(DirIn(DirIn(Current,DirName(h)),DirName(u)),FileName(蟋)), FileIn(DirIn(Current,DirName(qb)),FileName(sl)), FileIn(Current,FileName(i)))
[error]        is missing: FileIn(Current,FileName(c)) (QueryFileSpec.scala:31)
```

**Comment by drostron** [comment]
One more
```
[info] /data/fs should
[error]   ! MOVE view
[error]    java.net.BindException: Address already in use (Net.java:-2)
[error] sun.nio.ch.Net.bind0(Native Method)
[error] sun.nio.ch.Net.bind(Net.java:436)
[error] sun.nio.ch.Net.bind(Net.java:428)
[error] sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:214)
[error] org.http4s.blaze.channel.nio1.NIO1SocketServerGroup$$anonfun$bind$1.apply(NIO1SocketServerGroup.scala:63)
[error] org.http4s.blaze.channel.nio1.NIO1SocketServerGroup$$anonfun$bind$1.apply(NIO1SocketServerGroup.scala:62)
[error] org.http4s.blaze.channel.nio1.NIO1SocketServerGroup.bind(NIO1SocketServerGroup.scala:62)
[error] org.http4s.server.blaze.BlazeBuilder.org$http4s$server$blaze$BlazeBuilder$$$anonfun$2(BlazeServer.scala:180)
[error] org.http4s.server.blaze.BlazeBuilder$$Lambda$320/1893797734.apply(Unknown Source)
```

**Comment by mossprescott** [comment]
Seen multiple times yesterday and this morning, particularly on build ".3" (mongo read-only). Suspiciously, the hang seems to be during the query regression tests, some of which are expected to fail (and are marked as pending for the read-only backend).

The build seems to hang after running `ReadFileSpec`, then is eventually killed:
```
[info] 
[info] 
[info] Total for specification ReadFilesSpec
[info] Finished in 43 seconds, 227 ms
[info] 52 examples, 188 expectations, 0 failure, 0 error
[info] 
No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.
The build has been terminated
```

A good build of the same commit runs the Query tests at that point:
```
[info] 
[info] 
[info] Total for specification ReadFilesSpec
[info] Finished in 49 seconds, 379 ms
[info] 52 examples, 188 expectations, 0 failure, 0 error
[info] 
[info] ViewReadQueryRegressionSpec
[info] 
[info] mongodb_3_0 FileSystem should
```

**Comment by mossprescott** [comment]
I tried running a build with a hack in place to print when each Spec was _started_, thinking it would help to figure out which one(s) might be hanging. Of course that build didn't hang: https://travis-ci.org/mossprescott/quasar/jobs/150119716

What it did make clear is that we're starting _many_ of these in parallel. I can't tell how many of the actual tests are running concurrently, but a lot of the Specs are getting started at once before any of them produce any results. Maybe it's just that all of a project's Specs are created at once, I'm not sure.

**Comment by wemrysi** [comment]
Yeah, I think specs2 tries to run as many in parallel by default.

**Comment by mossprescott** [comment]
I wonder if that's causing problems by creating many simultaneous `MongoClient`s (each with its pool of threads and sockets). I could see that leading to unpredictable performance and/or hangs. And I don't see how there's much point in running more than one or two parallel tests in the Travis environment anyway. You would think SBT would provide simple settings to control this stuff.

**Comment by paulp** [comment]
```
[error]     x be 201 with dir
[error]      Gave up after only 1 passed tests. 10 tests were discarded. (DataServiceSpec.scala:54)
```

**Comment by paulp** [comment]
```
Warning: Failed to mount '/data/0/' because 'Connection failed: Timed out attempting to connect to: [localhost:27019]'.
```
[Then the wheels come off, a zillion failures](https://travis-ci.org/paulp/quasar/jobs/151255662).

**Comment by mossprescott** [comment]
Hopefully #1349 will resolve most of the MongoDB connection failures (which used to be hangs).

**Comment by mossprescott** [comment]
```
./scripts/build: line 79:  3058 Killed                  "$SBT" -DisIsolatedEnv=${ISOLATED_ENV:=false} clean checkHeaders coverage test exclusive:test coverageReport "set coverageEnabled := false" oneJar doc
The command "./scripts/build $MONGO_RELEASE" exited with 137.
before_cache.1
```
[Apparently](http://www.tldp.org/LDP/abs/html/exitcodes.html) 137 is the exit code when you `kill -9` a `bash` process (128+9). So I'm guessing Travis decided to kill this build and then just trapped the resulting error in a confusing way.

**Comment by paulp** [comment]
```
[info] Data Service should
[info]   GET
[info]     respond with NotFound
[error]       x if file does not exist
[error]        Falsified after 14 passed tests.
[error]        > ARG_0: FileIn(DirIn(DirIn(Root,DirName(1e)),DirName(g)),FileName(櫈7gZl4q/w%fd))
[error]        > ApiError(404 Path not found.,object[(path,"/1e/g/櫈7gZl4q$sep$w%FD")]) !== ApiError(404 Path not found.,object[(path,"/1e/g/櫈7gZl4q$sep$w%fd")]) (DataServiceSpec.scala:54)
```
The filename contains a `/` which it seems is encoded as `$sep$`.

**Comment by paulp** [comment]
Here's another similar one which failed a different job. I'm guessing these failures have existed a while and been masked somehow.
```
[info]     POST should
[error]       x be 400 with fs config and file path in X-File-Name header (13 seconds, 378 ms)
[error]        Falsified after 51 passed tests.
[error]        > ARG_0: FileIn(DirIn(DirIn(Current,DirName(nx.yQq.t)),DirName(0)),FileName(횜gx1톮w2/.mg젮ro.u%fbeanmwv4))
[error]        > ARG_1: DirIn(DirIn(DirIn(Root,DirName(e5f//.fk)),DirName(..)),DirName(IBehG))
[error]        > object[(path,"/e5f$sep$$sep$.fk/$dotdot$/IBehG/nx.yQq.t/0/횜gx1톮w2$sep$.mg젮ro.u%FBeanmwv4")] !== object[(path,"/e5f$sep$$sep$.fk/$dotdot$/IBehG/nx.yQq.t/0/횜gx1톮w2$sep$.mg젮ro.u%fbeanmwv4")] (MountServiceSpec.scala:43)
```

**Comment by paulp** [comment]
```
[info] /mount/fs should
[info]   + POST view (2 seconds, 983 ms)
[error]   ! PUT view (1 second, 472 ms)
[error]    java.net.BindException: Address already in use (Net.java:-2)
[error] sun.nio.ch.Net.bind0(Native Method)
[error] sun.nio.ch.Net.bind(Net.java:436)
[error] sun.nio.ch.Net.bind(Net.java:428)
[error] sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:214)
[error] org.http4s.blaze.channel.nio1.NIO1SocketServerGroup$$anonfun$bind$1.apply(NIO1SocketServerGroup.scala:63)
[error] org.http4s.blaze.channel.nio1.NIO1SocketServerGroup$$anonfun$bind$1.apply(NIO1SocketServerGroup.scala:62)
[error] org.http4s.blaze.channel.nio1.NIO1SocketServerGroup.bind(NIO1SocketServerGroup.scala:62)
[error] org.http4s.server.blaze.BlazeBuilder.org$http4s$server$blaze$BlazeBuilder$$$anonfun$2(BlazeServer.scala:180)
[error] org.http4s.server.blaze.BlazeBuilder$$Lambda$330/1967972333.apply(Unknown Source)
```

**Comment by jdegoes** [comment]
These are legitimate bugs that need to be fixed.

**Comment by mossprescott** [comment]
Weird one:

```
79344 [pool-8-thread-12] ERROR o.h.b.pipeline.Stage - Error running route: Request(method=MOVE, uri=/mount/fs/view/a, headers=Headers(Host: localhost:54119, Destination: /view/b, User-Agent: http4s-blaze/0.14.1a, Content-Length: 0)
java.lang.IndexOutOfBoundsException: 0
	at scala.collection.immutable.Vector.checkRangeConvert(Vector.scala:132)
	at scala.collection.immutable.Vector.apply(Vector.scala:122)
	at argonaut.PrettyParams$$anonfun$vectorMemo$1.apply(PrettyParams.scala:97)
	at argonaut.PrettyParams$$anonfun$vectorMemo$1.apply(PrettyParams.scala:92)
	at argonaut.PrettyParams.argonaut$PrettyParams$$lbrace$1(PrettyParams.scala:147)
	at argonaut.PrettyParams$$anonfun$argonaut$PrettyParams$$trav$1$6.apply(PrettyParams.scala:190)
	at argonaut.PrettyParams$$anonfun$argonaut$PrettyParams$$trav$1$6.apply(PrettyParams.scala:189)
	at argonaut.Json.fold(Json.scala:33)
	at argonaut.PrettyParams.argonaut$PrettyParams$$trav$1(PrettyParams.scala:171)
	at argonaut.PrettyParams.pretty(PrettyParams.scala:203)
	at argonaut.Json.pretty(Json.scala:420)
	at quasar.config.ConfigOps$class.asString(ConfigOps.scala:107)
	at quasar.config.WebConfig$$anon$3.asString(WebConfig.scala:27)
	at quasar.config.ConfigOps$class.quasar$config$ConfigOps$class$$$anonfun$20(ConfigOps.scala:95)
	at quasar.config.ConfigOps$class$$Lambda$3161/252971104.apply(Unknown Source)
	at scalaz.concurrent.Task$$anonfun$map$1$$anonfun$apply$2$$anonfun$apply$3.apply(Task.scala:42)
	at scalaz.concurrent.Task$.Try(Task.scala:448)
	at scalaz.concurrent.Task$$anonfun$map$1$$anonfun$apply$2.apply(Task.scala:42)
	at scalaz.concurrent.Task$$anonfun$map$1$$anonfun$apply$2.apply(Task.scala:42)
	at scalaz.$bslash$div.flatMap(Either.scala:135)
	at scalaz.concurrent.Task$$anonfun$map$1.apply(Task.scala:42)
	at scalaz.concurrent.Task$$anonfun$map$1.apply(Task.scala:42)
	at scala.Function1$$anonfun$andThen$1.apply(Function1.scala:52)
	at scalaz.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:58)
	at scalaz.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:58)
	at scalaz.concurrent.Future.step(Future.scala:119)
	at scalaz.concurrent.Future.unsafePerformListen(Future.scala:75)
	at scalaz.concurrent.Future$$anonfun$unsafePerformListen$1$$anonfun$apply$4.apply(Future.scala:79)
	at scalaz.concurrent.Future$$anonfun$unsafePerformListen$1$$anonfun$apply$4.apply(Future.scala:79)
	at scalaz.Free$$anonfun$map$1.apply(Free.scala:91)
	at scalaz.Free$$anonfun$map$1.apply(Free.scala:91)
	at scalaz.Free.resume(Free.scala:109)
	at scalaz.Free.go2$1(Free.scala:155)
	at scalaz.Free.go(Free.scala:159)
	at scalaz.Free.run(Free.scala:251)
	at scalaz.concurrent.Future$$anonfun$apply$15$$anon$3.call(Future.scala:432)
	at scalaz.concurrent.Future$$anonfun$apply$15$$anon$3.call(Future.scala:432)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[info] ServiceSpec
[info] 
[info] /mount/fs should
[info]   + POST view
[info]   + PUT view
[info]   + [SD-1833] replace view
[error]   x MOVE view
[error]    '\/-(404 Mount point not found.)' is not \/- with value'200 OK' (ServiceSpec.scala:162)
```

**Comment by paulp** [comment]
@mossprescott I've seen that before. I'm fairly sure it's a concurrency bug in argonaut.

**Comment by paulp** [comment]
(But I also think the argonaut bug is only being hit when it's trying to print the value which caused the other bug, so it's of secondary interest.)

**Comment by drostron** [comment]
[error]     x be 409 with dir to same location (746 ms)
[error]      Gave up after only 1 passed tests. 10 tests were discarded. (DataServiceSpec.scala:54)

**Comment by paulp** [comment]
```
 [info] ControlServiceSpec
 [info]
 [info] Control Service should
*[error]   ! restart on new port when PUT succeeds (37 seconds, 318 ms)
*[error]    java.util.concurrent.TimeoutException: Timed out after 30000 milliseconds (ControlServiceSpec.scala:56)
*[error] quasar.server.ControlServiceSpec.withServerExpectingRestart(ControlServiceSpec.scala:56)
*[error] quasar.server.ControlServiceSpec.quasar$server$ControlServiceSpec$$$anonfun$19(ControlServiceSpec.scala:74)
*[error] quasar.server.ControlServiceSpec$$Lambda$12354/650025269.apply(Unknown Source)
 [info]
*[error]   ! restart on default port when DELETE succeeds (37 seconds, 299 ms)
*[error]    java.util.concurrent.TimeoutException: Timed out after 30000 milliseconds (ControlServiceSpec.scala:56)
*[error] quasar.server.ControlServiceSpec.withServerExpectingRestart(ControlServiceSpec.scala:56)
*[error] quasar.server.ControlServiceSpec.quasar$server$ControlServiceSpec$$$anonfun$24(ControlServiceSpec.scala:84)
*[error] quasar.server.ControlServiceSpec$$Lambda$12355/1367383918.apply(Unknown Source)
```

**Comment by paulp** [comment]
And again.
```
[info] Control Service should
[error]   ! restart on new port when PUT succeeds (34 seconds, 488 ms)
[error]    java.util.concurrent.TimeoutException: Timed out after 30000 milliseconds (ControlServiceSpec.scala:56)
[error] quasar.server.ControlServiceSpec.withServerExpectingRestart(ControlServiceSpec.scala:56)
[error] quasar.server.ControlServiceSpec.quasar$server$ControlServiceSpec$$$anonfun$19(ControlServiceSpec.scala:74)
[error] quasar.server.ControlServiceSpec$$Lambda$11525/1392711101.apply(Unknown Source)
[info]
[error]   ! restart on default port when DELETE succeeds (34 seconds, 465 ms)
[error]    java.util.concurrent.TimeoutException: Timed out after 30000 milliseconds (ControlServiceSpec.scala:56)
[error] quasar.server.ControlServiceSpec.withServerExpectingRestart(ControlServiceSpec.scala:56)
[error] quasar.server.ControlServiceSpec.quasar$server$ControlServiceSpec$$$anonfun$24(ControlServiceSpec.scala:84)
[error] quasar.server.ControlServiceSpec$$Lambda$11526/1373815686.apply(Unknown Source)
```

**Comment by paulp** [comment]
```
[error]   x MOVE a directory containing views and files (1 second, 989 ms)
[error]    '\/-(404 Path not found.)' is not \/- with value'200 OK' (ServiceSpec.scala:244)
```

**Comment by paulp** [comment]
Here's the argonaut one again.
```
[info] postgresql FileSystem should
[info]   View Reads should
[error]     !  (4 seconds, 849 ms)
[error]      java.lang.IndexOutOfBoundsException: 0 (PrettyParams.scala:97)
[error] argonaut.PrettyParams$$anonfun$vectorMemo$1.apply(PrettyParams.scala:97)
[error] argonaut.PrettyParams$$anonfun$vectorMemo$1.apply(PrettyParams.scala:92)
[error] argonaut.PrettyParams.argonaut$PrettyParams$$lbrace$1(PrettyParams.scala:147)
[error] argonaut.PrettyParams$$anonfun$argonaut$PrettyParams$$trav$1$6.apply(PrettyParams.scala:190)
```

**Comment by paulp** [comment]
I opened argonaut-io/argonaut#243.

**Comment by mossprescott** [comment]
Another one in DataService that I don't think I've seen before: 
```
[error]     x download as zipped directory (32 seconds, 84 ms)
[error]      Gave up after only 2 passed tests. 12 tests were discarded. (QuasarSpecification.scala:31)
```

**Comment by mossprescott** [comment]
New one:

```
[info] MountConfigSpec
[info] 
[info] View config codec should
[info]   encode
[info]     + no vars (1 second, 2 ms)
[error]     ! with var (905 ms)
[error]      java.lang.IndexOutOfBoundsException: 0 (PrettyParams.scala:97)
[error] argonaut.PrettyParams$$anonfun$vectorMemo$1.apply(PrettyParams.scala:97)
[error] argonaut.PrettyParams$$anonfun$vectorMemo$1.apply(PrettyParams.scala:92)
[error] argonaut.PrettyParams.argonaut$PrettyParams$$lbrace$1(PrettyParams.scala:147)
[error] argonaut.PrettyParams$$anonfun$argonaut$PrettyParams$$trav$1$6.apply(PrettyParams.scala:190)
[error] argonaut.PrettyParams$$anonfun$argonaut$PrettyParams$$trav$1$6.apply(PrettyParams.scala:189)
[error] argonaut.Json.fold(Json.scala:33)
[error] argonaut.PrettyParams.argonaut$PrettyParams$$trav$1(PrettyParams.scala:171)
[error] argonaut.PrettyParams.pretty(PrettyParams.scala:203)
[error] argonaut.Json.nospaces(Json.scala:426)
[error] argonaut.Json.toString(Json.scala:468)
[error] argonaut.JsonScalazs$$anon$1.show(JsonScalaz.scala:62)
[error] argonaut.JsonScalazs$$anon$1.show(JsonScalaz.scala:50)
[error] argonaut.JsonScalazs$$anon$1.shows(JsonScalaz.scala:50)
```

**Comment by paulp** [comment]
@mossprescott Isn't that the same one you can see me posting a couple times in this issue?

**Comment by mossprescott** [comment]
@paulp: I don't see it up in that particular test, so I thought it was worth recording.

However, it also occurred in a different build on the same commit (after limiting parallel tasks), and this occurrence has been seen before:

```
[info] ViewReadQueryRegressionSpec
[info] 
[info] postgresql FileSystem should
[info]   View Reads should
[error]     !  (4 seconds, 153 ms)
[error]      java.lang.IndexOutOfBoundsException: 0 (PrettyParams.scala:97)
[error] argonaut.PrettyParams$$anonfun$vectorMemo$1.apply(PrettyParams.scala:97)
```

**Comment by paulp** [comment]
@mossprescott I think it can occur on any test. It's a race condition in argonaut.

**Comment by mossprescott** [comment]
That's worrying since we use argonaut pretty printing to produce essentially all of the quasar's output.

**Comment by paulp** [comment]
Another flaky test.
```
[info] TaskResource should
[error]   ! acquire once and release once
[error]    java.util.concurrent.TimeoutException: Timed out after 1000 milliseconds (file:1)
[info]
[error]   ! acquire once and release once with multiple concurrent requests
[error]    java.util.concurrent.TimeoutException: Timed out after 1000 milliseconds (file:1)
[info]
```

**Comment by drostron** [comment]
```
[error]   x lub symmetric (104 ms)
[error]    Falsified after 71 passed tests.
[error]    > ARG_0: Const(Obj(Map($date -> Str(Jan 1))))
[error]    > ARG_1: Coproduct(Const(NA),Const(Obj(Map($date -> Str(Jan 1)))))
[error]    > 'Coproduct(Const(NA),Const(Obj(Map($date -> Str(Jan 1)))))'
[error]    
[error]     is not equal to 
[error]    
[error]    'Const(Obj(Map($date -> Str(Jan 1))))' (QuasarSpecification.scala:31)
[error] Actual:   Co[product(Co]nst([NA),Const(]Ob...))[)]
[error] Expected: Co[]nst([]Ob...))[]
```

**Comment by drostron** [comment]
```
[error]   x round-trip all representable values (374 ms)
[error]    Gave up after only 18 passed tests. 92 tests were discarded. 
```

**Comment by drostron** [comment]
```
[error]   x have order-independent equality for arbitrary types (358 ms)
[error]    Falsified after 63 passed tests.
[error]    > ARG_0: Const(Obj(Map($date -> Str(Jan 1))))
[error]    > ARG_1: Coproduct(Const(NA),Const(Obj(Map($date -> Str(Jan 1)))))
[error]    > constant value Obj(Map($date -> Str(Jan 1))) !== constant value NA, or constant value Obj(Map($date -> Str(Jan 1))) (QuasarSpecification.scala:31)
```

**Comment by alissapajer** [comment]
```
[error]   x round-trip all flat rendered values that aren't "" (795 ms)
[error]    Falsified after 57 passed tests.
[error]    > ARG_0: Binary(Array[Byte](-29, 119, -70))
[error]    > 'Some(4.3E+7)' is Some but '4.3E+7' is not equal to '43e6' (QuasarSpecification.scala:31)
```
from
```
[error] Failed tests:
[error] 	quasar.main.PrettifySpecs
```

**Comment by alissapajer** [comment]
Would anyone object if each of these test failures were its own issue? Then they can be assigned and closed when fixed. We can tag them under the label "Build". (Or if that was meant for something else, a new label.)

**Comment by paulp** [comment]
@alissapajer All the ones which can be associated with a cause should certainly have their own ticket. I opened this one primarily because we were (and still are to some extent) plagued with failures without a known cause, like timeouts where nothing is logged and it's not even clear what was happening at the time.

**Comment by alissapajer** [comment]
Confirming the  race condition in argonaut still exists as of b7b6ea2041b77398772d1858e572b7f5768c4a3a.

**Comment by drostron** [comment]
```
[info] /mount/fs should
[error]   x POST view (1 second, 86 ms)
[error]    '\/-(500 Internal Server Error)' is not \/- with value'200 OK' (ServiceSpec.scala:90)
```

**Comment by alissapajer** [comment]
A lot of the scalacheck-related issues got fixed when scalacheck was last upgraded. So I agree with @zeeiyerWork's proposal above: if we have seen a test failure recently or can replicate it, it should get its own new issue. And then this issue can be closed.

**Comment by alissapajer** [comment]
Closing this issue. 

If we see any build failures, the protocol is to report each unique failure in its own issue rather than grouping  all failures in a single big issue.
